GoogleContainerTools/gcp-auth-webhook
# Added goroutine to watch namespaces and add secrets

Fixes a long overdue bug where newly created namespaces do not include the image registry pull secret. I created an [additional PR](https://github.com/kubernetes/minikube/pull/15403) in the minikube repo to go along with this one.

To test:
Ensure you have the minikube changes as well.
$ minikube addons enable gcp-auth
$ kubectl create ns new-namespace
$ kubectl get secrets -n new-namespace

Before:
```
<nothing>
```

After:
```
NAME       TYPE                      DATA   AGE
gcp-auth   kubernetes.io/dockercfg   1      10s
```

Additional Changes:
- Properly return from `updateCheck()` if it errors, instead of possibly crashing the server
- Updated dependencies
- Added more docs to README including how to run locally
